w3af/plugins/audit/open_api_auth.py

```diff
@@ -0,0 +1,372 @@
+            return params and auth.name in params and params[auth.name]
+
+        if auth.location == 'header' and freq.get_headers().iget(auth.name)[0]:
+            return True
+
+        return False
+
+    def _has_oauth2(self, freq):
+        """
+        Check if a fuzzable request contains OAuth2 info
+        if it's allowed by the API specification.
+
+        :param freq: The fuzzable request to be checked.
+        :return: True if the request contains OAuth2 info, False otherwise.
+        """
+        if not self._is_acceptable_auth_type(freq, 'oauth2'):
+            return False
+
+        return freq.get_headers().iget('Authorization', '')[0].startswith('Bearer')
+
+    def _is_acceptable_auth_type(self, freq, auth_type):
+        """
+        Check if the API specification allows a fuzzable request
+        to have a specific auth info.
+
+        :param freq: The fuzzable request to be checked.
+        :param auth_type: Auth method (oauth2, apiKey or basic).
+        :return: True if the API specification allows the request
+                 to have the specified auth method, False otherwise.
+        """
+        operation_id = self._get_operation_id(freq)
+        operation = self._get_operation_by_id(operation_id)
+        if not operation:
+            return False
+
+        for security_spec in operation.security_specs:
+            for key, value in security_spec.iteritems():
+                if key not in self._spec.security_definitions:
+                    # Should not happen.
+                    continue
+
+                security_definition = self._spec.security_definitions[key]
+                if security_definition.type == auth_type:
+                    return True
+
+        return False
+
+    def get_long_desc(self):
+        """
+        :return: A DETAILED description of the plugin functions and features.
+        """
+        return """
+        This plugin checks that REST API endpoints require authentication
+        according to their API specification.
+
+        OpenAPI specification offers the following structures
+        to define requirements for authentication:
+          * 'securityDefinitions' section describes all available authentication mechanisms.
+            It offers three methods: basic, apiKey and oauth2.
+          * 'security' section may be used in each operation to specify
+            which authentication mechanisms the operation accepts.
+          * Global 'security' section describes authentication mechanisms
+            for operations which don't have their own 'security' section
+
+        Currently the check is pretty simple:
+          * Remove authentication info from the request (API key, header, etc).
+          * Send the modified request to the endpoint.
+          * Check if the response has 401 error code (access denied).
+
+        A couple of important notes:
+          * The plugin requires the 'crawl.open_api' plugin to be enabled.
```

Comment:
You can force that. Plugins can have other plugins as dependencies.
https://github.com/andresriancho/w3af/blob/master/w3af/core/controllers/plugins/plugin.py#L123

Reply:
When plugin A depends on B, and plugin A is enabled, plugin B will be automatically enabled too.

Reply:
That's nice, I didn't know about it. Let me add such a dependency.